Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -213,48 +242,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate {
 
 extension SearchViewController: UITableViewDataSource {
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
-
-        if let currentSection = SearchListSection(rawValue: indexPath.section) {
-            switch currentSection {
-            case .SearchSuggestions:
-                cell.textLabel?.text = searchSuggestions[indexPath.row]
-                cell.detailTextLabel?.text = nil
-                cell.imageView?.image = nil
-                cell.isAccessibilityElement = false
-                cell.accessibilityLabel = nil
-            case .BookmarksAndHistory:
-                if let site = data[indexPath.row] as? Site {
-                    cell.textLabel?.text = site.title
-                    cell.detailTextLabel?.text = site.url
-                    if let img = site.icon? {
-                        let imgUrl = NSURL(string: img.url)
-                        cell.imageView?.sd_setImageWithURL(imgUrl, placeholderImage: self.profile.favicons.defaultIcon)
-                    } else {
-                        cell.imageView?.image = self.profile.favicons.defaultIcon
-                    }
-                    cell.isAccessibilityElement = false
-                    cell.accessibilityLabel = nil
+        switch SearchListSection(rawValue: indexPath.section)! {
```

Comment:
> If the rawValue: gets an invalid value, the app will crash. Better to not show anything then instead I think.

Is it? Either way, there's a bug in the app that we need to fix. If it crashes right away, we immediately notice and get a stack trace of the exact failure location. If we paper over it, we might not notice as soon, and when we do it will take longer to track down. Same thing with implicitly unwrapped optionals that must be set in classes: we could make them all optional and check for cases where we forgot to assign them, but it's easier to be sure things are working when we just assume they're set.

I like to use assertions liberally for the same reason. As useful as testing is, it's nearly impossible to get code coverage for every nook and cranny. Having these built-in checks helps to prevent broken edge cases from slipping by unnoticed.